pkg/ensurance/collector/cadvisor/cadvisor_linux.go

```diff
@@ -25,69 +24,59 @@ import (
 	"github.com/gocrane/crane/pkg/utils"
 )
 
-const (
-	cadvisorCollectorName = "cadvisor"
-)
-
 func init() {
-	registerMetrics(cadvisorCollectorName, []types.MetricName{types.MetricNameContainerCpuTotalUsage, types.MetricNameContainerSchedRunQueueTime}, NewCadvisor)
+	//registerMetrics(cadvisorCollectorName, []types.MetricName{types.MetricNameContainerCpuTotalUsage, types.MetricNameContainerSchedRunQueueTime}, NewCadvisor)
 }
 
-type CgroupState struct {
+type ContainerState struct {
 	stat      cadvisorapiv2.ContainerInfo
 	timestamp time.Time
 }
 
 //CadvisorCollector is the collector to collect container state
 type CadvisorCollector struct {
-	Manager   cmanager.Manager
+	manager   cmanager.Manager
 	podLister corelisters.PodLister
+	name      types.CollectType
 
-	cgroupState           map[string]CgroupState
-	MemCache              *cmemory.InMemoryCache
-	SysFs                 csysfs.SysFs
-	IncludeMetrics        cadvisorcontainer.MetricSet
-	MaxHousekeepingConfig cmanager.HouskeepingConfig
+	latestContainersStates map[string]ContainerState
 }
 
-func NewCadvisor(context *NodeLocalContext) (nodeLocalCollector, error) {
+func NewCadvisor(podLister corelisters.PodLister) types.Collector {
 
 	var includedMetrics = cadvisorcontainer.MetricSet{
 		cadvisorcontainer.CpuUsageMetrics:         struct{}{},
 		cadvisorcontainer.ProcessSchedulerMetrics: struct{}{},
 	}
 
-	var allowDynamic bool = true
-	var maxHousekeepingInterval time.Duration = 10 * time.Second
+	var allowDynamic = true
```

Comment:
short format `allowDynamic := true` is better when you define a variable with value.